### ISSUE TITLE
Pointer pressure support in drawing

### ISSUE THREAD

**Issue by Krastanov** [fmeringdal/nettu-meet#11]
I am prompted to make this issue (feature request) as a follow up on a comment made on the [HN annoucement](https://news.ycombinator.com/item?id=27439519)

I currently use a wacom tablet when teaching and whiteboards that support pressure sensitivity (xournal++) provide much better experience and quality of handwriting than those that support only pressed/not pressed state (zoom, google jamboard).

It seems that if browser/hardware support is not a consideration then [pressure.js](https://pressurejs.com/) is all one needs. However, both in firefox and chrome there are missing implementations that make this useless for standard desktop drawing tablets like wacom (but it works great with less standard hardware like the apple pen). Here for instance is the firefox bug report describing how [pressure is generally fixed to 0.5 instead of the actual value reported by the tablet](https://bugzilla.mozilla.org/show_bug.cgi?id=1487509). Here is the meta bug listing [all pen input issues](https://bugzilla.mozilla.org/show_bug.cgi?id=1631377). Lastly, here is a test page where you can double [check what pressure is reported in your browser](https://patrickhlauke.github.io/touch/tests/event-listener-force-pressure.html).

So, I guess my question after all of this is: is there a timeline for when I will be able to use a wacom tablet with firefox (or chrome) in a pressure-sensitive nettu-meet?

**Comment by Krastanov** [comment]
P.S. Either way, this is a fantastic piece of software and I am thankful you have created it and open-sourced it.

**Comment by fmeringdal** [comment]
Thanks for the detailed description on existing libraries / issues.
This is **without a doubt** the most important feature request for Nettu Meet. 
I will have to do some research on this and get back to your within 1-2 days for a more detailed roadmap. But if I were to guess this should be implemented within 1 or 2 weeks if some of the existing pressure libraries out there actually works (which I am sure some of them do).

**Comment by miroslavpejic85** [comment]
Interesting! 👀

**Comment by fmeringdal** [comment]
Hi @Krastanov ,

Initially I was thinking that it was not possible to use the Nettu Meet whiteboard with a wacom tablet, but it seems like this is working and that the only problem is that it does not support pressure? Can you confirm that you are able to make drawings with the tablet?

Also regarding the pressure sensitivity support, I would really like to hear more about your the use cases for having a pressure enabled whiteboard as I am not a tablet user yet (going to buy one very soon).
Like is it just the thickness of the line when drawing, or are there other use cases as well?

Thanks!

**Comment by Krastanov** [comment]
Yes, you are right! Sorry for the confusion. Later today I will attach pictures of what I meant when I said handwriting without pressure sensitivity is worse.

basically "mouse << tablet < tablet with pressure sensitivity"

**Comment by Krastanov** [comment]
It was not later today, but here are the examples:

This is xournal++ which has very basic pressure sensitivity:

The ends of strokes when writing letters are thinner, and by varying the pressure you can make a shape with a thick border and shade it with lighter hachure.
![image](https://user-images.githubusercontent.com/705248/122956436-fff3a700-d34e-11eb-9767-fc46a15774fa.png)
![image](https://user-images.githubusercontent.com/705248/122956700-3af5da80-d34f-11eb-9511-9c48729a8ad6.png)

Google jamboard and Zoom try to fake some of this with smoothing and to be fair their are pretty ok, but any variation in thickness here is from some RNG that has little to do with actual pressure. Varying thickness of writing is just not a thing here.
![image](https://user-images.githubusercontent.com/705248/122957423-ed2da200-d34f-11eb-9d20-433b52bfcaa6.png)
![image](https://user-images.githubusercontent.com/705248/122957532-059dbc80-d350-11eb-8ca2-36a7ac999268.png)

For a single line of text this does not matter much. For a one-hour long lecture with a lot of circling and underlining and shading of graphs, it becomes very noticeable

**Comment by Krastanov** [comment]
I understand that initially there was a misunderstanding of what this feature request was about. Feel free to close this as a low priority, especially given the amount of browser bugs reported above.

**Comment by fmeringdal** [comment]
Thanks for taking the time to explain this. 
This is not going to have such a high priority because of the lack of browser support, but I will leave it open so that hopefully it can be solved in the not too distant future :)
